### ISSUE TITLE
expose more settings to user control in `TransformersModelManager`

### ISSUE THREAD

**Issue by iulusoy** [ssciwr/moralization#89]
The following settings could be exposed to the user:
- use a different tokenizer than shipped with the model
- tokenizer kwargs
- data collator kwargs
- evaluation metric (currently only seqeval, see [here](https://huggingface.co/docs/evaluate/choosing_a_metric) for some options)
- `batch_size` in the dataloader, currently set to 8
- more kwargs for the optimizer other than learning rate
- schedular options that handle the adjustment of the learning rate over the epochs (`scheduler_name` currently linear, `num_train_epochs` currently set to 3, also possible to change `num_warmup_steps` that is currently set to 0), see [here](https://huggingface.co/docs/timm/reference/schedulers) for more options and [here](https://www.kaggle.com/code/rhtsingh/guide-to-huggingface-schedulers-differential-lrs) for more information
